src/api/procedures/modifyMultiSig.ts

```diff
@@ -0,0 +1,154 @@
+import BigNumber from 'bignumber.js';
+
+import { PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, MultiSig, Signer, TxTags } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import { signerToSignatory, signerToString, stringToAccountId } from '~/utils/conversion';
+import { checkTxType } from '~/utils/internal';
+
+/**
+ * @hidden
+ */
+export interface ModifyMultiSigParams {
+  /**
+   * The MultiSig to be modified
+   */
+  multiSig: MultiSig;
+  /**
+   * The signers to set for the MultiSig
+   */
+  signers: Signer[];
+}
+
+export interface Storage {
+  signersToAdd: Signer[];
+  signersToRemove: Signer[];
+  requiredSignatures: BigNumber;
+}
+
+/**
+ * @hidden
+ */
+function calculateSignerDelta(
+  current: Signer[],
+  target: Signer[]
+): Pick<Storage, 'signersToAdd' | 'signersToRemove'> {
+  const currentSet = new Set(current.map(signerToString));
+  const targetSet = new Set(target.map(signerToString));
+
+  const newSigners = new Set([...target].filter(s => !currentSet.has(signerToString(s))));
+  const removedSigners = new Set([...current].filter(s => !targetSet.has(signerToString(s))));
+
+  return { signersToAdd: Array.from(newSigners), signersToRemove: Array.from(removedSigners) };
+}
+
+/**
+ * @hidden
+ */
+export async function prepareModifyMultiSig(
+  this: Procedure<ModifyMultiSigParams, void, Storage>,
+  args: ModifyMultiSigParams
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: { tx },
+    },
+    storage: { signersToAdd, signersToRemove, requiredSignatures },
+    context,
+  } = this;
+  const { signers, multiSig: multi } = args;
+
+  const [signingIdentity, creator] = await Promise.all([
+    context.getSigningIdentity(),
+    multi.getCreator(),
+  ]);
+
+  if (signersToAdd.length === 0 && signersToRemove.length === 0) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'The given signers are equal to the current signers',
+    });
+  }
+
+  if (!creator.isEqual(signingIdentity)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'A MultiSig can only be modified by its creator',
+    });
+  }
+
+  const rawAddress = stringToAccountId(multi.address, context);
+
+  if (requiredSignatures.gt(signers.length)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'The number of required signatures should not exceed the number of signers',
+    });
+  }
+
+  const transactions = [];
+
+  if (signersToAdd.length > 0) {
+    const rawAddedSigners = signersToAdd.map(s => signerToSignatory(s, context));
```

Comment:
```suggestion
    const rawAddedSigners = signersToAdd.map(signer => signerToSignatory(signer, context));
```
I think we avoid using single character variable names. Can change to `signer` or `newSigner` or `signerToAdd`